pkg/processor/eventsource/eventsource.go

```diff
@@ -61,13 +61,19 @@ func (aes *AbstractEventSource) GetKind() string {
 func (aes *AbstractEventSource) SubmitEventToWorker(event nuclio.Event,
 	timeout time.Duration) (response interface{}, submitError error, processError error) {
 
+	var workerInstance *worker.Worker
+
 	defer func() {
 		if err := recover(); err != nil {
 			aes.Logger.ErrorWith("Panic caught during submit events", "err", err)
 
 			response = nil
 			submitError = fmt.Errorf("Panic caught during submit events: %s", err)
 			processError = nil
+
+			if workerInstance != nil {
```

Comment:
This should be outside the `if`, we'd like to release the worker even when there was no error.

Reply:
The worker is released outside the if: https://github.com/nuclio/nuclio/pull/129/files/356540b3c4a78addcdce81e08bd0e4f4c78e567f#diff-8bef7883f39f9b8a1240f9b68ef94f2bR95

Am I missing something?

Reply:
Nope, missed that.